src/ArbitrumFoundationVestingWallet.sol

```diff
@@ -0,0 +1,99 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity 0.8.16;
+
+import "@openzeppelin/contracts-upgradeable/finance/VestingWalletUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol";
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+
+import {IERC20VotesUpgradeable} from "./Util.sol";
+
+interface IL2ArbitrumGoverner {
+    function token() external view returns (IERC20VotesUpgradeable);
+    function EXCLUDE_ADDRESS() external view returns (address);
+}
+
+/**
+ * @notice A wallet for foundation owned founds as per AIP-1.1 specification.
+ * DAO can migrate funds to new wallet.
+ * Wallet vests funds over time on a linear schedule.
+ * Governance votes are delegated to exclude address.
+ * Beneficiary can be updated by DAO.
+ */
+
+contract ArbitrumFoundationVestingWallet is VestingWalletUpgradeable, OwnableUpgradeable {
+    using SafeERC20 for IERC20;
+
+    address private _beneficiary;
+
+    constructor() {
+        _disableInitializers();
+    }
+
+    function initialize(
+        address _beneficiaryAddress,
+        uint64 _startTimestamp,
+        uint64 _durationSeconds,
+        address _arbitrumGoverner,
+        address _owner
```

Comment:
We so far only use the address registry for gov action contracts, and I think I'd prefer to keep the separation (ie., this one would be a bit awkward since if we deploy the L2 address registry for this, the ArbitrumFoundationWallet couldn't be included, which it probably should be)